google/cloud/storage/blob.py

```diff
@@ -2759,26 +2759,19 @@ def upload_from_file(
             "md5", "crc32c" and None. The default is None.
 
         :type retry: google.api_core.retry.Retry or google.cloud.storage.retry.ConditionalRetryPolicy
-        :param retry: (Optional) How to retry the RPC. A None value will disable
-            retries. A google.api_core.retry.Retry value will enable retries,
-            and the object will define retriable response codes and errors and
-            configure backoff and timeout options.
-
-            A google.cloud.storage.retry.ConditionalRetryPolicy value wraps a
-            Retry object and activates it only if certain conditions are met.
-            This class exists to provide safe defaults for RPC calls that are
-            not technically safe to retry normally (due to potential data
-            duplication or other side-effects) but become safe to retry if a
-            condition such as if_generation_match is set.
+        :param retry: (Optional) How to retry the RPC.
+            The default value is ``DEFAULT_RETRY_IF_GENERATION_SPECIFIED``, which
```

Comment:
There are some grammar issues with the first paragraph. May I suggest something like the following:

The default value is ``DEFAULT_RETRY_IF_GENERATION_SPECIFIED``, a conditional retry policy which will only enable retries if ``if_generation_match`` or ``generation`` is set, in order to ensure requests are idempotent before retrying them. Change the value to ``DEFAULT_RETRY`` or another ``google.api_core.retry.Retry`` object to enable retries regardless of generation setting.

See [Configuring Retries](https://cloud.google.com/python/docs/reference/storage/latest/retry_timeout) for more information.